variants/generator/generate.py

```diff
@@ -0,0 +1,660 @@
+#!/usr/bin/env python3
```

Comment:
Why not write the generator in Go? ;)

It's actually a serious question, the fewer languages the lower the learning curve :)

Reply:
Also, a comment about what this file does would be nice. It's pretty complicated..

Reply:
I did think of writing it in Go, and also putting it in a separate project. The main problem is my lack of familiarity with Go, but also the out-of-the-box features in Python are much more powerful. I think most of the work here was getting the algorithm right (and I'm still not sure that it is, as I've only run one test case variant).

Documentation is definitely needed, and I think there's also some dead code in here. I wanted to get something checked in.

Reply:
Cool, no sweat :) A Python script is 100% better than no script, so it's no biggie. But it would be even better if using the same language as the rest of the code :D